### DISCUSSION TITLE
Is it possible to change the wifi mac address from software on the PicoW?

### DISCUSSION THREAD

**Discussion by the-maldridge** [earlephilhower/arduino-pico#2390] [Q&A]
Title says it all.  I've perused the code and see its possible to read the MAC, but not methods to set it.  Does the CY43 driver expose this as something that it would be possible to set?  I want to be able to encode the device ID in an administratively assigned MAC address to make it easier to pre-allocate DHCP to devices, but that is of course dependent on being able to know the MAC in advance.

**Comment by earlephilhower** [comment]
Not now, now.  The code in [LwipIntfDev.h](https://github.com/earlephilhower/arduino-pico/blob/master/libraries/lwIP_Ethernet/src/LwipIntfDev.h) uses the method The RPI team used to make one up using the flash ID, and after that it's modified to make it [legal and unique](https://github.com/earlephilhower/arduino-pico/blob/bc7adf42c2f4fc6e448bb31c21cef07e658e9289/libraries/lwIP_Ethernet/src/LwipIntfDev.h#L355-L367) in the device when multiple Ethernet devices are present.

A new template method in LwipIntfDev could be used to override that (and then you'd need to pull that up thru the `WiFi` class I believe) to make it end user accessible.
